src/ai/backend/storage/vfs/__init__.py

```diff
@@ -254,8 +254,10 @@ def _scandir(target_path: Path, q: janus._SyncQueueProxy[Sentinel | DirEntry]) -
                         if entry.is_dir() and not entry.is_symlink():
                             next_paths.append(Path(entry.path))
```

Comment:
I think `break if not recursive` should come here, because the current implementation depends on the order that `os.scandir()` returns the directory entries.